DSpace/DSpace
# Add Canvas metadata to IIIF seeAlso annotations

## References
* Related to [DSpace#8811](https://github.com/DSpace/DSpace/pull/8811)
## Description
This PR updates the existing `seeAlso` annotation lists to use `bitstream.iiif.canvasid` metadata that was introduced in #8811. Previously, canvas IDs were generated dynamically based on bundle sequence. #8811 derives the canvasid from the bitstream UUID. It also adds the ability to link IIIF annotations with an associated canvas via the `bitstream.iiif.canvasid` metadata field. 

This PR:

- Provides a new CLI script to add `bitstream.iiif.canvasid` metadata to the OCR `Bitstream` records in the "OtherContent" `Bundle`. 
- Updates the `AnnotationListService` to return the UUID-based `Canvas` Id in the IIIF `seeAlso` annotation list response.
- Includes integration Tests

Sample annotation response with Canvas Id (in the `on` property):

```
{
@id: "http://localhost:8080/server/iiif/4f04ac7c-dfa0-4e09-9ded-b9a1f1c2c473",
@type: "oa:Annotation",
motivation: "oa:linking",
resource: {
@id: "http://localhost:8080/server/api/core/bitstreams/4f04ac7c-dfa0-4e09-9ded-b9a1f1c2c473/content",
label: "0048.xml",
format: "text/xml"
},
on: {
@id: "http://localhost:8080/server/iiif/add816e6-3c43-411c-acd3-5758007f29d2/canvas/18e0b3b2-29a6-4da6-9354-0d99f10d9010",
@type: "sc:Canvas"
},
within: [ ]
},
```

(The PR also includes changes to Solr query parameters in `WordHighlightSolrSearch`.  This PR is not about updating search but I choose to slip these into the PR on the recommendation of the [plugin](https://dbmdz.github.io/solr-ocrhighlighting/0.8.3/) authors. They do not otherwise impact the code. _Since it now appears this PR will not make it into 7.6 I'll create a new tiny PR for this change._)


## Instructions for Reviewers

List of changes in this PR:
* AnnotationLinkCLI:
    * New CLI script that adds canvasid metadata to OCR files. (No Angular UI version as of now but this can be added).
* AnnotationLinkService:
    * Processes Communities, Collections, or Items. 
    * Auto-detects Alto, hOcr and MiniOcr file formats in the "OtherContent" `Bundle`.
    * Supports adding, deleting or updating metadata.
* AnnotationListService:
    *  Updated annotations to include an `on` property containing the canvasid of the corresponding image `Bitstream`.

To test this:

1. Create an  IIIF-enabled Item with one or more OCR files in the "OtherContent" `Bundle` (the OCR format can be based on Alto, hOcr or MinOcr) and one or more images in the default `Bundle`. Note that as before, the number and sequence of ORC files and images matters. 
2. Open the IIIF Manifest and follow the `seeAlso` link. You should see an annotation list with entries that do NOT have the `on` property and canvasid.
3. Run the  `iiif-ocr-canvas-link` script with item identifier `-i` and eperson email `-e`
4. Reload the `seeAlso` annotation list. You should see the canvasid in the `on` property (as shown above).

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
